lib/resty/kafka/producer.lua

```diff
@@ -189,6 +200,18 @@ local function _batch_send(self, sendbuffer)
         -- aggregator
         local send_num, sendbroker = sendbuffer:aggregator(self.client)
         if send_num == 0 then
+            local count = 0
+            for topic, partitions in pairs(sendbuffer.topics) do
+                for partition_id, buffer in pairs(partitions) do
+                    count = count + sendbuffer.topics[topic][partition_id].index / 2
+                end
+            end
+            if count ~= 0 and (try_num == self.max_retry) then
+                -- we have messages to send, but cannot do it due lack of metadata,
+                -- so we need to trigger error_handler to not to loose them
+                return false
+            end
+
```

Comment:
I'm not really understand this changed, seems it doesn't change anything:)
`break` will terminate the loop, and `return nil` (Lua default behavior) means the same as `return false`.